### PR TITLE
Add C# type test example (pattern matching)

### PR DESCRIPTION
**Pull request by 31**
Add a simple example, `GetNode("MySprite") is Sprite2D mySprite`. This is a very common suggestion in the C# Discord channel and quite popular in modern C#, and adding it to the doc makes the doc much more useful as a reference.

It happens a lot that someone sends a link to this page, but also has to mention "and if you use `is Foo f`, then it's even better for this situation!"

There's more text that could be added to this section explaining how it works, maybe analogizing to `as`+`if not null` or explicitly describing how it relates to the `is` section earlier. But I think it's good enough to have a brief description, an example, and keep the C# doc link to point anyone who wants more info in the right direction.

### CODE REVIEW

File: tutorials/scripting/c_sharp/c_sharp_features.rst [hunk 1]
```diff
@@ -81,8 +81,19 @@ otherwise it returns true. Note that using the ``is`` operator against ``null``
         // This block can never happen.
     }
 
-For more advanced type checking, you can look into `Pattern Matching <https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching>`_.
+**Type test with variable declaration**
```
Comment (raulsntos) [inline tutorials/scripting/c_sharp/c_sharp_features.rst:84]: I'm not sure this needs to be a new section, seems like it could be part of the previous one.

Comment (31) [reply tutorials/scripting/c_sharp/c_sharp_features.rst]: Hmm. Yeah, reworded and put the "advanced" note back where it was.

**Review by raulsntos — APPROVED** [review]
LGTM. Thanks for contributing to the C# documentation!

File: tutorials/scripting/c_sharp/c_sharp_features.rst [hunk 2]
```diff
@@ -81,6 +81,17 @@ otherwise it returns true. Note that using the ``is`` operator against ``null``
         // This block can never happen.
     }
 
+You can add a variable declaration to conditionally store the result of the cast
+if the ``is`` operator returns true.
```
Comment (YuriSizov) [inline tutorials/scripting/c_sharp/c_sharp_features.rst:85]:
```suggestion
You can also declare a new variable to conditionally store the result of the cast
if the ``is`` operator returns ``true``.
```

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**